spec/takeaway_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'takeaway'
+
+describe Takeaway do
+  
+  let(:fake_menu) { double(:fake_menu, list: [{name: "Chicken", quantity: 1, price: 4.5}]) }
+  let(:menu_class) { double(:menu_class, new: fake_menu) }
+  let(:fake_sum) { double(:fake_sum) }
+  let(:sum_class) { double(:sum_class, new: fake_sum) }
+  let(:fake_text) { double(:fake_text, message: true) }
+  let(:text_class) { double(:text_class, new: fake_text) }
+  subject(:takeaway) { described_class.new(menu_class: menu_class, sum_class: sum_class, text_class: text_class) }
+
+    describe '#initialize' do
+      it 'shows the menu' do
+        expect(takeaway.menu).to eq fake_menu
+      end
+      it 'has a sum' do
+        expect(takeaway.sum).to eq fake_sum
+      end
+      it 'returns an empty basket' do
+        expect(takeaway.basket).to eq []
+      end
+      it 'initializes text' do
+        expect(takeaway.text).to eq fake_text
+      end
+    end
+
+    describe '#menu_list' do
+      context 'when viewing menu' do
+        it 'returns menu' do
+          expect(fake_menu).to receive(:list)
+          takeaway.menu_list
+        end
+      end
+    end
+
+    describe '#order' do
+      context 'when adding a dish to the basket' do
+        it 'adds a dish' do
+          expect { takeaway.order('Chicken') }.to change { takeaway.basket.count }
```

Comment:
This is better - testing behaviour rather than state. Your assertion isn't quite testing the expected outcome however. When we add 'Chicken', we expect the dish in the basket to be 'Chicken'. We should test that the basket contains 'Chicken', not merely that the count has changed.